bin/dry-run.rb

```diff
@@ -482,11 +482,23 @@ def security_fix?(dependency)
     end
   end
 
-  latest_allowed_version = checker.vulnerable? ?
-    checker.lowest_resolvable_security_fix_version :
-    checker.latest_resolvable_version
+  latest_allowed_version = if checker.vulnerable?
+                             checker.lowest_resolvable_security_fix_version
+                           else
+                             checker.latest_resolvable_version
+                           end
   puts " => latest allowed version is #{latest_allowed_version || dep.version}"
 
+  parents = checker.blocking_parent_dependencies
+  if parents.any?
+    puts " => The update is not possible because of the following conflicting "\
+      "dependencies:"
+  end
+  parents.each do |parent|
+    puts "   #{parent['name']} (#{parent['version']}) which requires:"
```

Comment:
I contemplated returning `Dependency` objects instead, but those are already used for lot's of different things and would end up having lots of unused fields. We could introduce a new class for this as well. I ended up opting for a hash instead, so that we can pass it through to the calling code without having to map over it, saving us having to transform the object back and forth. wdyt?

Reply:
SGTM. the documented keys of the hash were 💯 for bridging the gap between "a completely untyped hash" and "a new class", awesome!

Reply:
Do we need to check this every time or could we nest this under `if requirements_to_unlock == :update_not_possible`?